packages/emitter-framework/src/java/components/enum.tsx

```diff
@@ -0,0 +1,57 @@
+import { Enum } from "@typespec/compiler";
+import * as jv from  "@alloy-js/java"
+import { mapJoin } from "@alloy-js/core";
+import { TypeExpression } from "../../typescript/index.js";
+
+
+export interface EnumProps {
+  type: Enum;
+}
+
+export function EnumDeclaration({ type }: EnumProps) {
+
+  const members = Array.from(type.members.values());
+
+  let valueType: string | undefined;
+  let isConsistent = true;
+
+  members.forEach((member, index) => {
+    if (member.value !== undefined) {
+      const currentType = typeof member.value === "string" ? "String" : "int";
+
+      if (index === 0) {
+        valueType = currentType;
+      } else if (valueType !== currentType) {
+        isConsistent = false;
+      }
+    }
+  });
+
+  if (!isConsistent) {
+    throw new Error("All enum values must have the same type");
+  }
+
+  let constructor = null;
+  let variable = null;
+  if (valueType) {
+    variable = <jv.Variable type={valueType} name="customValue" private={true}/>;
+    constructor = <jv.Constructor name={type.name} parameters={{"customValue": valueType}}
+                                  children={"this.customValue = customValue;"} private={true}/>;
+  }
+
+  const joinedMembers = mapJoin(members, (member) => {
+    const customValue = member.value ? <jv.Value value={member.value}/> : null;
```

Comment:
Where you have it returning `null` if there is no value, set it to `'null'` as if you are emitting the java `null`. At the moment it's not declaring a value for enums without a value and the java code isn't valid.

![image](https://github.com/user-attachments/assets/809d013b-28e5-485b-b3d5-bb24e21e748e)

Reply:
Actually you can probably just do
```ts
const customValue = <jv.Value value={member.value} />;
```
As the value component will render `null` if member.value is undefined anyway